src/docker/docker_build.rs

```diff
@@ -0,0 +1,93 @@
+use crate::common::exec;
+use crate::docker::ImageUri;
+use anyhow::Result;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use tokio::process::Command;
+
+/// Can execute a `docker build` command. This follows the builder pattern, for example:
+///
+/// ```
+/// let build = DockerBuild.dockerfile("./Dockerfile").context(".").execute().await?;
+/// ```
+pub(crate) struct DockerBuild {
+    dockerfile: Option<PathBuf>,
+    context_dir: PathBuf,
+    tag: Option<ImageUri>,
```

Comment:
Docker images *can* have multiple tags. Does it make sense for us to expose that?